src/fetchServices.ts

```diff
@@ -0,0 +1,189 @@
+    res.data.services.map(item => {
+      if (item.serviceName) {
+        serviceNames.push(item.serviceName);
+      }
+    });
+  }
+
+  if (res.data.nextPageToken) {
+    await listServices(client, serviceNames, res.data.nextPageToken);
+  }
+  return serviceNames;
+}
+
+/**
+ * Make a call to the Google Cloud Service Management API's services.getConfig endpoint
+ * @param serviceName - the hostname of a service ('foo.googleapis.com')
+ * @return the service config of the passed hostname
+ */
+export async function getServiceConfig(
+  serviceName: string
+): Promise<servicemanagement_v1.Schema$Service> {
+  const auth = await google.auth.getClient({
+    scopes: ['https://www.googleapis.com/auth/cloud-platform'],
+  });
+  const servicemanagement = google.servicemanagement({
+    version: 'v1',
+    auth: auth,
+  });
+  const res = await servicemanagement.services.getConfig(
+    {
+      serviceName: serviceName,
+    },
+    {
+      retryConfig: {
+        retry: 10,
+      },
+    }
+  );
+  return res.data;
+}
+
+/**
+ * Get service configs for all available services.
+ * @return the service configuration object
+ */
+export async function getAllServiceConfigs() {
+  const serviceNames = await getAllServiceNames();
+  const serviceConfigs = serviceNames.map(async (name: string) => {
+    await getServiceConfig(name);
+  });
+  return serviceConfigs;
+}
+
+/**
+ * Determines if a given service is a 'Cloud API'.
+ * @param serviceName - the hostname of a service ('foo.googleapis.com')
+ * @return well is it, or not?
+ */
+export async function isCloudApi(serviceName: string): Promise<boolean> {
+  const serviceConfig = await getServiceConfig(serviceName);
+
+  if (serviceConfig.title?.includes('Firebase')) {
+    return false;
+  }
+  if (serviceConfig.authentication?.rules) {
+    const scopes: boolean[] = serviceConfig.authentication.rules.map(item => {
+      if (item.oauth?.canonicalScopes) {
+        const scopesStr: string = item.oauth.canonicalScopes;
+        if (scopesStr.includes('auth/firebase')) {
+          return false;
+        } else if (scopesStr.includes('auth/cloud-platform')) {
+          return true;
+        } else if (scopesStr.includes('auth/drive')) {
+          return true;
+        } else if (scopesStr.includes('auth/apps')) {
+          return true;
+        }
+      }
+      return false;
+    });
+
+    if (scopes.includes(true)) {
+      return true;
+    }
+  }
+
+  // usage: { requirements: [ 'serviceusage.googleapis.com/tos/cloud' ] }
+  if (serviceConfig.usage?.requirements) {
+    const usage: string[] = serviceConfig.usage.requirements;
+    if (usage.includes('serviceusage.googleapis.com/tos/cloud')) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+/**
+ * Classify all available services as Cloud API or not and export to a known Google Sheet.
+ */
+export async function exportApisToSheets() {
+  const services: string[] = await Promise.all(await getAllServiceNames());
```

Comment:
That doesn't smell right.  `getAllServiceNames()` already returns a single promise which can be await-ed.  I don't know what the `Promise.all` here is supposed to do.